jest.config.js

```diff
@@ -0,0 +1,32 @@
+module.exports = {
+    collectCoverageFrom: [
+        '**/*.{js,jsx,ts,tsx}',
+        '!**/*.d.ts',
+        '!**/node_modules/**',
+    ],
+    moduleNameMapper: {
+        /* Handle CSS imports (with CSS modules)
+        https://jestjs.io/docs/webpack#mocking-css-modules */
+        '^.+\\.module\\.(css|sass|scss)$': 'identity-obj-proxy',
+
+        // Handle CSS imports (without CSS modules)
+        '^.+\\.(css|sass|scss)$': '<rootDir>/__mocks__/styleMock.js',
+
+        /* Handle image imports
+        https://jestjs.io/docs/webpack#handling-static-assets */
+        '^.+\\.(jpg|jpeg|png|gif|webp|svg)$': '<rootDir>/__mocks__/fileMock.js',
+    },
+    preset: 'ts-jest',
+    setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
+    testPathIgnorePatterns: ['<rootDir>/node_modules/', '<rootDir>/.next/'],
+    testEnvironment: 'jsdom',
```

Comment:
can we make this `['node', 'jsdom']` because we also need to test the sign in sign out api (or we can specify the test env on top of thoose files)